testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/federation/OrganizationFederationTest.java

```diff
@@ -0,0 +1,130 @@
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+
+public class OrganizationFederationTest extends AbstractOrganizationTest {
+
+    private static final File CONFIG_DIR = new File(System.getProperty("auth.server.config.dir", ""));
+
+    @Before
+    public void onBefore() {
+        createOrganization("orga");
+        createOrganization("orgb");
+    }
+
+    @After
+    public void onAfter() {
+        List<UserRepresentation> users = testRealm().users().search("member");
+
+        if (!users.isEmpty()) {
+            UserRepresentation member = users.get(0);
+            testRealm().users().get(member.getId()).remove();
+        }
+    }
+
+    @Override
+    public void importTestRealms() {
+        super.importTestRealms();
+        log.infof("Test realms imported");
+
+        afterImportTestRealm();
+    }
+
+    protected void afterImportTestRealm() {
+        copyPropertiesFiles();
+        final String propertyFile = CONFIG_DIR.getAbsolutePath() + File.separator + "user-password.properties";
+        testingClient.server().run(session -> {
+
+            RealmModel appRealm = session.realms().getRealmByName(AbstractLDAPTest.TEST_REALM_NAME);
+
+            // add user-prop provider with lower priority
+            ComponentModel userPropProvider = new ComponentModel();
+            userPropProvider.setName("user-props");
+            userPropProvider.setProviderId(UserPropertyFileStorageFactory.PROVIDER_ID);
+            userPropProvider.setProviderType(UserStorageProvider.class.getName());
+            userPropProvider.setConfig(new MultivaluedHashMap<>());
+            userPropProvider.getConfig().putSingle("priority", Integer.toString(0));
+            userPropProvider.getConfig().putSingle("propertyFile", propertyFile);
+            userPropProvider.getConfig().putSingle("federatedStorage", "true");
+            appRealm.addComponentModel(userPropProvider);
+        });
+    }
+
+    private void copyPropertiesFiles() throws RuntimeException {
+        try {
+            // copy files used by the following user-props user provider
+            File stResDir = new File(getClass().getResource("/storage-test").toURI());
+            if (stResDir.exists() && stResDir.isDirectory() && CONFIG_DIR.exists() && CONFIG_DIR.isDirectory()) {
+                for (File f : stResDir.listFiles()) {
+                    log.infof("Copying %s to %s", f.getName(), CONFIG_DIR.getAbsolutePath());
+                    FileUtils.copyFileToDirectory(f, CONFIG_DIR);
+                }
+            } else {
+                throw new RuntimeException("Property `auth.server.config.dir` must be set to run the test.");
+            }
+        } catch (IOException | RuntimeException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testGetByMember() {
```

Comment:
It would also be nice to test authentication and claims mapping (the `organization` scope and claims) so that we have coverage for these core capabilities when the user is federated.

If you want I can contribute the tests. They should be similar to:

* https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/authentication/OrganizationAuthenticationTest.java#L50
* https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/mapper/OrganizationOIDCProtocolMapperTest.java#L84